pymc3/distributions/continuous.py

```diff
@@ -1654,6 +1655,93 @@ def logcdf(self, value):
         )
 
 
+class AsymmetricLaplace(Continuous):
+    r"""
+    Asymmetric-Laplace log-likelihood.
+
+    The pdf of this distribution is
+
+    ..math::
+        {f(x|\\b,\kappa) =
+            \left({\frac{\\b}{\kappa + 1/\kappa}}\right)\,e^{-(x)\\b\,s\kappa ^{s}}}
+        where s = sgn(x)
+
+    Parameters
+    ----------
+    b:
+        Scale parameter (b > 0)
+    kappa:
+        Symmetry parameter (kappa > 0)
+
+    See also: https://en.wikipedia.org/wiki/Asymmetric_Laplace_distribution
+    """
+
+    def __init__(self, b, kappa, testval=0.0, *args, **kwargs):
+        self.b = tt.as_tensor_variable(b)
+        self.kappa = tt.as_tensor_variable(kappa)
+
+        super().__init__(*args, **kwargs, testval=testval)
+
+    def _random(self, b, kappa, size=None):
+        if size is not None:
+            u = np.random.uniform(size=size)
+            x = -np.log((1 - u) * (1 + kappa ** 2)) / (kappa * b) * (
+                u > ((kappa ** 2) / (1 + kappa ** 2))
+            ) + kappa * np.log(u * (1 + kappa ** 2) / (kappa ** 2)) / b * (
+                u < ((kappa ** 2) / (1 + kappa ** 2))
+            )
+            return x
+
+        u = np.random.uniform()
+        if u > (kappa ** 2) / (1 + kappa ** 2):
+            x = -np.log((1 - u) * (1 + kappa ** 2)) / (kappa * b)
+        else:
+            x = kappa * np.log(u * (1 + kappa ** 2) / (kappa ** 2)) / b
+
```

Comment:
The formulas are not very readable. I would suggest to go explicit mapping the random draws with their corresponding domains -

```suggestion
        u = np.random.uniform(size=size)
        switch = kappa ** 2 / (1 + kappa ** 2)
        non_positive_x = kappa * np.log(u * (1 / switch)) / b
        positive_x = -np.log((1 - u) * (1 + kappa ** 2)) / (kappa * b)
        draws = non_positive_x * (u <= switch) + positive_x * (u > switch)
        return draws
```
Then there will be no need of these if else conditionals.